controllers/apim/ratelimitpolicy_controller.go

```diff
@@ -85,15 +100,343 @@ func (r *RateLimitPolicyReconciler) Reconcile(eventCtx context.Context, req ctrl
+						"envoy_grpc": map[string]string{
+							// It should be (outbound|8081||limitador.kuadrant-system.svc.cluster.local) but
+							// limitador rejects request if '|' is present in the cluster name.
+							"cluster_name": PatchedLimitadorClusterName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	patchRaw, _ := json.Marshal(patchUnstructured)
+
+	prePatch := networkingv1alpha3.EnvoyFilter_Patch{}
+	err := prePatch.UnmarshalJSON(patchRaw)
+	if err != nil {
+		panic(err)
+	}
+	postPatch := prePatch.DeepCopy()
+	postPatch.Value.Fields["name"] = &types.Value{
+		Kind: &types.Value_StringValue{
+			StringValue: "envoy.filters.http.postauth.ratelimit",
+		},
+	}
+	// update domain
+	postPatch.Value.Fields["typed_config"].GetStructValue().Fields["domain"] = &types.Value{
+		Kind: &types.Value_StringValue{
+			StringValue: "postauth",
+		},
+	}
+	// update stage
+	postPatch.Value.Fields["typed_config"].GetStructValue().Fields["stage"] = &types.Value{
+		Kind: &types.Value_NumberValue{
+			NumberValue: postAuthRLStage,
+		},
+	}
+
+	preAuthFilterPatch := networkingv1alpha3.EnvoyFilter_EnvoyConfigObjectPatch{
+		ApplyTo: networkingv1alpha3.EnvoyFilter_HTTP_FILTER,
+		Match: &networkingv1alpha3.EnvoyFilter_EnvoyConfigObjectMatch{
+			Context: networkingv1alpha3.EnvoyFilter_GATEWAY,
+			ObjectTypes: &networkingv1alpha3.EnvoyFilter_EnvoyConfigObjectMatch_Listener{
+				Listener: &networkingv1alpha3.EnvoyFilter_ListenerMatch{
+					PortNumber: 8080, // Kuadrant-gateway listens on this port by default
+					FilterChain: &networkingv1alpha3.EnvoyFilter_ListenerMatch_FilterChainMatch{
+						Filter: &networkingv1alpha3.EnvoyFilter_ListenerMatch_FilterMatch{
+							Name: "envoy.filters.network.http_connection_manager",
+							SubFilter: &networkingv1alpha3.EnvoyFilter_ListenerMatch_SubFilterMatch{
+								Name: "envoy.filters.http.router",
+							},
+						},
+					},
+				},
+			},
+		},
+		Patch: &prePatch,
+	}
+
+	postAuthFilterPatch := preAuthFilterPatch
+	postAuthFilterPatch.Patch = postPatch
+
+	// Eventually, this should be dropped since it's a temp-fix for Kuadrant/limitador#53
+	clusterPatch := networkingv1alpha3.EnvoyFilter_EnvoyConfigObjectPatch{
+		ApplyTo: networkingv1alpha3.EnvoyFilter_CLUSTER,
+		Match: &networkingv1alpha3.EnvoyFilter_EnvoyConfigObjectMatch{
+			Context: networkingv1alpha3.EnvoyFilter_GATEWAY,
+			ObjectTypes: &networkingv1alpha3.EnvoyFilter_EnvoyConfigObjectMatch_Cluster{
+				Cluster: &networkingv1alpha3.EnvoyFilter_ClusterMatch{
+					Name: DefaultLimitadorClusterName,
+				},
+			},
+		},
+		Patch: &networkingv1alpha3.EnvoyFilter_Patch{
+			Operation: networkingv1alpha3.EnvoyFilter_Patch_MERGE,
+			Value: &types.Struct{
+				Fields: map[string]*types.Value{
+					"name": {
+						Kind: &types.Value_StringValue{
+							StringValue: PatchedLimitadorClusterName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	factory := istioprovider.EnvoyFilterFactory{
+		ObjectName: gateway.Name + "-ratelimit-filters",
```

Comment:
I wonder if the envoyfilter resource name is linked to the rate limiting policy, more than  to the gateway

Reply:
ratelimit filters patch is required once per gateway so if it's per RLP then we'll have redundant patches to the same gateway if multiple RLPs target same gateway.